### ISSUE TITLE
Spyder3 segfaults Fedora-packaged PyQt5

### ISSUE THREAD

**Issue by milindsmart** [spyder-ide/spyder#4678]
## Description of your problem

With Fedora-packaged PyQt5 and pip (user)-installed Spyder, spyder segfaults, unless I also install PyQt5 in pip

**What steps will reproduce the problem?**

1. `sudo dnf install python3-qt5` and almost all other spyder dependencies
2. `pip3 install --user spyder` which installs the dependencies still left from (1)
3. `spyder3`

**What is the expected output? What do you see instead?**
`Segmentation fault (core dumped)` on terminal. Nothing on graphical launch.

**Please provide any additional information below**
Even with PyQt5 from pip3, it still requires adding `from OpenGL import GL` to spyder start.py (https://github.com/spyder-ide/spyder/issues/3332#issuecomment-254249729), else again it segfaults. With PyQt5 installed from pip3, the splash screen does appear and then goes off around, I think, "Setting up main window".

## Versions and main components

* Spyder Version: 3.1.4
* Python Version: 3.5.3
* Qt Version: 5.7.1
* PyQt Version: 5.7.4 (system), 5.8.2 (pip3)
* Operating system: Fedora 25 x64

## Dependencies

Spyder does not start

**Comment by milindsmart** [comment]
Exactly this issue. https://unix.stackexchange.com/questions/369665/pip-saves-the-day-on-fedora-25-for-no-module-named-pyqt5-qtwebenginewidgets-er

**Comment by ccordoba12** [comment]
So Fedora Qt5 package comes without `QtWebEngineWidgets`? I don't understand how that's something we can solve in Spyder and why it's note better to complain with Fedora maintainers...

**Comment by ccordoba12** [comment]
Regarding the pip packages, please complain with Phil Thompson, the PyQt5 maintainer. Again, it's not something we can solve in Spyder because we don't create those packages nor we are related to PyQt5 in any way.

However, we test Spyder with the PyQt5 wheels in virtual machines that run Ubuntu 14.04 and it runs perfectly fine there. So I'd say the problem is specific to Fedora.

**Comment by ccordoba12** [comment]
Sorry but I'm closing this issue. We can't do anything else than what we already did with `pyopengl`.

It's very unfortunate that `from OpenGL import GL` doesn't work for you.

**Comment by milindsmart** [comment]
I'm sorry, looks like I wasn't clear:

- With `pip3 install --user pyqt5` AND with `from OpenGL import GL` : Spyder works
- With `pip3 install --user pyqt5` AND without `from OpenGL import GL`: `[1:1:0100/000000:ERROR:broker_posix.cc(41)] Invalid node channel message
Segmentation fault (core dumped)`
- Without `pip3 install --user pyqt5` AND with `from OpenGL import GL`: `Segmentation fault (core dumped)`
- Without `pip3 install --user pyqt5` AND without `from OpenGL import GL` : `Segmentation fault (core dumped)`

> So Fedora Qt5 package comes without QtWebEngineWidgets? 

Maybe. It would be great if you could help me confirm that is the case, in which case I'd be happy to raise it with Fedora maintainers. Does the presence/absence of a `.so` file confirm it?

**Comment by ccordoba12** [comment]
So at least things work under some circumstances, great to know that our fix will be really valuable!

Sorry, I can't help you with the Fedora part, but I really think you should raise the case with them, at least to know what they know about it.

**Comment by milindsmart** [comment]
Sorry, it was my mistake, `sudo dnf install python3-qt5-webengine` changes things in the previous table:

- With `pip3 install --user pyqt5` AND with `from OpenGL import GL` : Spyder works
- With `pip3 install --user pyqt5` AND without `from OpenGL import GL`: `[1:1:0100/000000:ERROR:broker_posix.cc(41)] Invalid node channel message
Segmentation fault (core dumped)`
- Without `pip3 install --user pyqt5` AND with `from OpenGL import GL`: `Segmentation fault (core dumped)`
- Without `pip3 install --user pyqt5` AND with `sudo dnf install python3-qt5-webengine` AND with `from OpenGL import GL`: Spyder works
- Without `pip3 install --user pyqt5` AND without `from OpenGL import GL` : `Segmentation fault (core dumped)`

Also, one more piece of good news is that `sudo dnf install python3-pyopengl` works as well as `pip3 install --user pyopengl`.  

To summarise for recent Fedora users who want to minimise local pip installs: 

- `sudo dnf install` Qt5, Qt5 WebEngine, Python3-PyQt5, Python3-Qt5-WebEngine, Jupyter core and client, Python3-PyOpenGL, IPyKernel, IPython-genutils
- `pip3 install --user ipython==5 notebook`
- Apply `from OpenGL import GL`
- Run `spyder3`

Thanks for the help.

**Comment by ccordoba12** [comment]
Great!! Thanks for your thorough investigation.
